File: Sources/CountriesFlagsObjC/source/Countries/UIImage+Seychelles.m
//
//  UIImage+Seychelles.m
//  CountriesFlags
//

#import "Countries/UIImage+Seychelles.h"

@implementation UIImage (Seychelles)

+ (UIImage * _Nullable)flagSeychellesWithSize:(CGSize)size {
    UIColor* colorBlue = [UIColor colorWithRed:0.0f
                                         green:61.0f/255.0f
                                          blue:136.0f/255.0f
                                         alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:252.0f/255.0f
                                           green:217.0f/255.0f
                                            blue:85.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorRed = [UIColor colorWithRed:215.0f/255.0f
                                        green:35.0f/255.0f
                                         blue:35.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorWhite = [UIColor whiteColor];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:123.0f/255.0f
                                           blue:58.0f/255.0f
                                          alpha:1.0f];
    
    UIGraphicsBeginImageContextWithOptions(size, YES, 0.0f);
    
    [colorBlue setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorYellow setFill];
    UIBezierPath* path = [UIBezierPath bezierPath];
    [path moveToPoint:CGPointMake(size.width / 6.0f * 2.0f, 0.0f)];
    [path addLineToPoint:CGPointMake(size.width / 6.0f * 4.0f, 0.0f)];
    [path addLineToPoint:CGPointMake(0.0f, size.height)];
    [path closePath];
    [path fill];
    
    [colorRed setFill];
    [path removeAllPoints];
    [path moveToPoint:CGPointMake(size.width / 6.0f * 4.0f, 0.0f)];
    [path addLineToPoint:CGPointMake(size.width, 0.0f)];
    [path addLineToPoint:CGPointMake(size.width, size.height / 3.0f)];
    [path addLineToPoint:CGPointMake(0.0f, size.height)];
    [path closePath];
    [path fill];

    [colorWhite setFill];
    [path removeAllPoints];
    [path moveToPoint:CGPointMake(size.width, size.height / 3.0f)];
    [path addLineToPoint:CGPointMake(size.width, size.height / 3.0f * 2.0f)];
    [path addLineToPoint:CGPointMake(0.0f, size.height)];
    [path closePath];
    [path fill];

    [colorGreen setFill];
    [path removeAllPoints];
    [path moveToPoint:CGPointMake(size.width, size.height / 3.0 * 2.0)];
    [path addLineToPoint:CGPointMake(size.width, size.height)];
    [path addLineToPoint:CGPointMake(0.0f, size.height)];
    [path closePath];
    [path fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+Seychelles.swift
//
//  UIImage+Seychelles.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Seychelles "wikipedia"
     
     Generates the flag of Seychelles.
     For more information see [wikipedia].

     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Seychelles.
     */
    internal static func flagSeychelles(with size: CGSize) -> UIImage? {
        let colorBlue = UIColor(red: 0.0,
                                green: 61.0/255.0,
                                blue: 136.0/255.0,
                                alpha: 1.0)
        let colorYellow = UIColor(red: 252.0/255.0,
                                  green: 217.0/255.0,
                                  blue: 85.0/255.0,
                                  alpha: 1.0)
        let colorRed = UIColor(red: 215.0/255.0,
                               green: 35.0/255.0,
                               blue: 35.0/255.0,
                               alpha: 1.0)
        let colorWhite = UIColor.white
        let colorGreen = UIColor(red: 0.0,
                                 green: 123.0/255.0,
                                 blue: 58.0/255.0,
                                 alpha: 1.0)

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorBlue.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))
        
        colorYellow.setFill()
        var path = UIBezierPath()
        path.move(to: CGPoint(x: size.width / 6.0 * 2.0, y: 0.0))
        path.addLine(to: CGPoint(x: size.width / 6.0 * 4.0, y: 0.0))
        path.addLine(to: CGPoint(x: 0.0, y: size.height))
        path.close()
        path.fill()
        
        colorRed.setFill()
        path = UIBezierPath()
        path.move(to: CGPoint(x: size.width / 6.0 * 4.0, y: 0.0))
        path.addLine(to: CGPoint(x: size.width, y: 0.0))
        path.addLine(to: CGPoint(x: size.width, y: size.height / 3.0))
        path.addLine(to: CGPoint(x: 0.0, y: size.height))
        path.close()
        path.fill()
        
        colorWhite.setFill()
        path = UIBezierPath()
        path.move(to: CGPoint(x: size.width, y: size.height / 3.0))
        path.addLine(to: CGPoint(x: size.width, y: size.height / 3.0 * 2.0))
        path.addLine(to: CGPoint(x: 0.0, y: size.height))
        path.close()
        path.fill()
        
        colorGreen.setFill()
        path = UIBezierPath()
        path.move(to: CGPoint(x: size.width, y: size.height / 3.0 * 2.0))
        path.addLine(to: CGPoint(x: size.width, y: size.height))
        path.addLine(to: CGPoint(x: 0.0, y: size.height))
        path.close()
        path.fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
